test/controllers/user_controller_test.rb

```diff
@@ -665,67 +668,72 @@ def test_lost_password
     assert_equal 1, email.to.count
-    assert_equal users(:normal_user).email, email.to.first
+    assert_equal user.email, email.to.first
     ActionMailer::Base.deliveries.clear
 
     # Test resetting using an address that matches a different user
     # that has the same address in a different case
     assert_difference "ActionMailer::Base.deliveries.size", 1 do
-      post :lost_password, :user => { :email => users(:normal_user).email.upcase }
+      post :lost_password, :user => { :email => user.email.upcase }
     end
     assert_response :redirect
     assert_redirected_to :action => :login
     assert_match /^Sorry you lost it/, flash[:notice]
     email = ActionMailer::Base.deliveries.first
     assert_equal 1, email.to.count
-    assert_equal users(:uppercase_user).email, email.to.first
+    assert_equal uppercase_user.email, email.to.first
     ActionMailer::Base.deliveries.clear
 
     # Test resetting using an address that is a case insensitive match
     # for more than one user but not an exact match for either
     assert_no_difference "ActionMailer::Base.deliveries.size" do
-      post :lost_password, :user => { :email => users(:normal_user).email.titlecase }
+      post :lost_password, :user => { :email => user.email.titlecase }
     end
     assert_response :success
     assert_template :lost_password
     assert_select ".error", /^Could not find that email address/
 
     # Test resetting using the address as recorded for a user that has an
     # address which is case insensitively unique
+    third_user = create(:user)
     assert_difference "ActionMailer::Base.deliveries.size", 1 do
-      post :lost_password, :user => { :email => users(:public_user).email }
+      post :lost_password, :user => { :email => third_user.email }
     end
     assert_response :redirect
     assert_redirected_to :action => :login
     assert_match /^Sorry you lost it/, flash[:notice]
     email = ActionMailer::Base.deliveries.first
     assert_equal 1, email.to.count
-    assert_equal users(:public_user).email, email.to.first
+    assert_equal third_user.email, email.to.first
     ActionMailer::Base.deliveries.clear
 
     # Test resetting using an address that matches a user that has the
     # same (case insensitively unique) address in a different case
     assert_difference "ActionMailer::Base.deliveries.size", 1 do
-      post :lost_password, :user => { :email => users(:public_user).email.upcase }
+      post :lost_password, :user => { :email => third_user.email.upcase }
     end
     assert_response :redirect
     assert_redirected_to :action => :login
     assert_match /^Sorry you lost it/, flash[:notice]
     email = ActionMailer::Base.deliveries.first
     assert_equal 1, email.to.count
-    assert_equal users(:public_user).email, email.to.first
+    assert_equal third_user.email, email.to.first
     ActionMailer::Base.deliveries.clear
   end
 
   def test_reset_password
+    user = create(:user)
```

Comment:
This needs a `:pending` as it was using the inactive user before and part of the test is checking that doing a password reset on a pending user activates the user as a side-effect.